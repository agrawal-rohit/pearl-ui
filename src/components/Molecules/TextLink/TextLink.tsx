import React from "react";
import Text from "../../Atoms/Text/Text";
import { useMolecularComponentConfig } from "../../../hooks/useMolecularComponentConfig";
import { useColorScheme } from "../../../hooks/useColorScheme";
import Pressable, { PressableProps } from "../../Atoms/Pressable/Pressable";

type TextLinkProps = PressableProps & {
  /** Size of the text link. */
  size?: string;
  /** Variant of the text link. */
  variant?: string;
  /** Active color palette of the text link */
  colorScheme?: string;
};

/** TextLink wraps a Text component with a Pressable component that can be used to trigger an action or event, such as submitting a form, opening a dialog, canceling an action, or performing a delete operation */
const TextLink: React.FC<TextLinkProps> = ({
  children,
  colorScheme = "primary",
  isDisabled = false,
  ...props
}) => {
  let multiComponentStyles = useMolecularComponentConfig("TextLink", props, {
    size: props["size"],
    variant: props["variant"],
  });
  if (colorScheme !== "primary") {
    multiComponentStyles = useColorScheme(colorScheme, multiComponentStyles);
  }

  return (
    <Pressable
      isDisabled={isDisabled}
      opacity={isDisabled ? 0.5 : 1}
      onPress={props.onPress}
      accessibilityLabel={
        props.accessibilityLabel ? props.accessibilityLabel : children
      }
      accessibilityState={{ disabled: isDisabled }}
      isDisabledAndroidRipple
      {...multiComponentStyles.root}
    >
      <Text {...multiComponentStyles.text}>{children}</Text>
    </Pressable>
  );
};

export default TextLink;
