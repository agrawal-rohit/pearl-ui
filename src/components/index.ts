// Atoms
export { default as Box, BoxProps } from "./atoms/box/box";
export { default as Center } from "./atoms/center/center";
export { default as Collapse, CollapseProps } from "./atoms/collapse/collapse";
export { default as Divider, DividerProps } from "./atoms/divider/divider";
export { default as Fade, FadeProps } from "./atoms/fade/fade";
export { default as Grid, GridProps } from "./atoms/grid/grid";
export { default as Icon, IconProps } from "./atoms/icon/icon";
export {
  default as Pressable,
  PressableProps,
} from "./atoms/pressable/pressable";
export {
  default as ScaleFade,
  ScaleFadeProps,
} from "./atoms/scale-fade/scale-fade";
export {
  default as List,
  OrderedList,
  UnorderedList,
  ListProps,
} from "./atoms/list/list";
export { default as Screen, ScreenProps } from "./atoms/screen/screen";
export { default as Skeleton, SkeletonProps } from "./atoms/skeleton/skeleton";
export {
  default as SkeletonCircle,
  SkeletonCircleProps,
} from "./atoms/skeleton/skeleton-circle";
export {
  default as SkeletonText,
  SkeletonTextProps,
} from "./atoms/skeleton/skeleton-text";
export { default as Slide, SlideProps } from "./atoms/slide/slide";
export {
  default as SlideFade,
  SlideFadeProps,
} from "./atoms/slide-fade/slide-fade";
export { default as Spacer } from "./atoms/spacer/spacer";
export { default as Spinner, SpinnerProps } from "./atoms/spinner/spinner";
export {
  default as Stack,
  HStack,
  VStack,
  ZStack,
  HStackProps,
  VStackProps,
  StackProps,
} from "./atoms/stack/stack";
export { default as Text, TextProps } from "./atoms/text/text";

// Molecules
export { default as Avatar, AvatarProps } from "./molecules/avatar/avatar";
export {
  default as AvatarGroup,
  AvatarGroupProps,
} from "./molecules/avatar/avatar-group";
export { default as Badge, BadgeProps } from "./molecules/badge/badge";
export {
  default as withBadge,
  WithBadgeOptions,
} from "./molecules/badge/withBadge";
export { default as Button, ButtonProps } from "./molecules/button/button";
export {
  default as ButtonGroup,
  ButtonGroupProps,
} from "./molecules/button/button-group";
export {
  default as CheckBox,
  CheckBoxProps,
} from "./molecules/checkbox/checkbox";
export {
  default as CheckBoxGroup,
  CheckBoxGroupProps,
} from "./molecules/checkbox/checkbox-group";
export {
  default as IconButton,
  IconButtonProps,
} from "./molecules/icon-button/icon-button";
export { default as Image, ImageProps } from "./molecules/image/image";
export { default as Input, InputProps } from "./molecules/input/input";
export {
  default as PinInput,
  PinInputProps,
} from "./molecules/pin-input/pin-input";
export {
  default as Progress,
  ProgressProps,
} from "./molecules/progress/progress";
export { default as Radio, RadioProps } from "./molecules/radio/radio";
export {
  default as RadioGroup,
  RadioGroupProps,
} from "./molecules/radio/radio-group";
export { default as Switch, SwitchProps } from "./molecules/switch/switch";
export {
  default as Textarea,
  TextareaProps,
} from "./molecules/textarea/textarea";
export { default as Video, VideoProps } from "./molecules/video/video";
export {
  default as Accordion,
  AccordionProps,
  AccordionItemProps,
  AccordionButtonProps,
  AccordionPanelProps,
  AccordionItem,
  AccordionButton,
  AccordionPanel,
} from "./molecules/accordion/accordion";
export { default as Alert, AlertProps } from "./molecules/alert/alert";
