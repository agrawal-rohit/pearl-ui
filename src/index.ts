export { default as Box } from "./components/Box/Box";
export { default as Button } from "./components/Button/Button";
export { default as Text } from "./components/Text/Text";
