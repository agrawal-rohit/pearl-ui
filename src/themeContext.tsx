import React, { useEffect, useState, useContext, createContext } from "react";
import { useColorScheme } from "react-native";
import { ThemeProvider as RestyleThemeProvider } from "@shopify/restyle";
import { baseTheme, darkTheme, Theme } from "./theme";
import { ReactElement } from "react-color/node_modules/@types/react";

type ThemeType = "light" | "dark" | "system";

interface IThemeContext {
  themeMode: ThemeType;
  toggleTheme(theme: string): void;
}

interface ThemeProviderProps {
  defaultMode?: ThemeType;
  light?: Theme;
  dark?: Theme;
}

export const themeContext = createContext({} as IThemeContext);

export const ThemeProvider: React.FC<ThemeProviderProps> = ({
  defaultMode = "light",
  light = baseTheme,
  dark = darkTheme,
  children,
}) => {
  const [themeMode, setThemeMode] = useState<ThemeType>(defaultMode);
  const systemThemeStyle = useColorScheme() as ThemeType;

  const toggleTheme = () => {
    if (themeMode === "light") {
      setThemeMode("dark");
    } else {
      setThemeMode("light");
    }
  };

  useEffect(() => {
    // Set theme
    if (defaultMode === "system") {
      setThemeMode(systemThemeStyle);
    }
  }, []);

  return (
    <themeContext.Provider value={{ themeMode, toggleTheme } as IThemeContext}>
      <RestyleThemeProvider theme={themeMode === "light" ? light : dark}>
        {children}
      </RestyleThemeProvider>
    </themeContext.Provider>
  );
};

export const useTheme = () => useContext(themeContext);
