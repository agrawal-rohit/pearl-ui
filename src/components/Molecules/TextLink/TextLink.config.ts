import { MolecularComponentConfig } from "../../../theme/src/types";
import { TextLinkProps } from "./TextLink";

export default {
  parts: ["root", "text"],
  baseStyle: {
    root: {
      activeOpacity: 0.8,
    },
    text: {
      color: "primary.500",
    },
  },
  sizes: {
    xs: {
      text: {
        variant: "btn4",
      },
    },
    s: {
      text: {
        variant: "btn3",
      },
    },
    m: {
      text: {
        variant: "btn2",
      },
    },
    l: {
      text: {
        variant: "btn1",
      },
    },
  },
  defaults: {
    size: "m",
  },
} as MolecularComponentConfig<TextLinkProps>;
