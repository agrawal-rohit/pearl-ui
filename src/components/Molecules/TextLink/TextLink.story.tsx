import React from "react";
import { action } from "@storybook/addon-actions";
import { storiesOf } from "@storybook/react-native";
import TextLink from "./TextLink";
import Screen from "../../Atoms/Screen/Screen";

storiesOf("TextLink", module)
  .addDecorator((getStory) => <Screen>{getStory()}</Screen>)
  .add("Sizes", () => (
    <>
      <TextLink onPress={action("clicked-text")} size="xs">
        Button text
      </TextLink>

      <TextLink onPress={action("clicked-text")} size="s">
        Button text
      </TextLink>

      <TextLink onPress={action("clicked-text")} size="m">
        Button text
      </TextLink>

      <TextLink onPress={action("clicked-text")} size="l">
        Button text
      </TextLink>
    </>
  ))
  .add("Disabled", () => (
    <>
      <TextLink onPress={action("clicked-text")} isDisabled>
        Button text
      </TextLink>
    </>
  ))
  .add("Color Scheme", () => (
    <>
      <TextLink onPress={action("clicked-text")}>Button text</TextLink>
      <TextLink onPress={action("clicked-text")} colorScheme="secondary">
        Button text
      </TextLink>
    </>
  ));
