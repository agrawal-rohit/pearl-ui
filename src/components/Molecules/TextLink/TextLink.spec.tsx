import React from "react";
import TextLink from "./TextLink";
import { fireEvent, render } from "@testing-library/react-native";
import { ThemeProvider } from "../../../theme/src/themeContext";

jest.useFakeTimers();

describe("Molecules/TextLink", () => {
  it("passes the snapshot test for basic setup", async () => {
    const onPress = jest.fn();

    const tree = await render(
      <ThemeProvider>
        <TextLink onPress={onPress}>Text Link press</TextLink>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("passes the snapshot test for different sizes", async () => {
    const onPress = jest.fn();

    const tree = await render(
      <ThemeProvider>
        <TextLink onPress={onPress} size="xs">
          Text Link press
        </TextLink>
        <TextLink onPress={onPress} size="s">
          Text Link press
        </TextLink>
        <TextLink onPress={onPress} size="m">
          Text Link press
        </TextLink>
        <TextLink onPress={onPress} size="l">
          Text Link press
        </TextLink>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("passes the snapshot test for different color schemes", async () => {
    const tree = await render(
      <ThemeProvider>
        <TextLink colorScheme="primary">Primary text link</TextLink>
        <TextLink colorScheme="secondary">Secondary text link</TextLink>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("passes the snapshot test when using style props", async () => {
    const onPress = jest.fn();

    const tree = await render(
      <ThemeProvider>
        <TextLink
          onPress={onPress}
          backgroundColor="primary.500"
          borderColor="neutral.400"
          borderWidth={2}
        >
          Text link press
        </TextLink>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("captures the onPress event", async () => {
    const onPress = jest.fn();

    const { getByTestId } = await render(
      <ThemeProvider>
        <TextLink onPress={onPress} testID="testOnPress">
          Button press
        </TextLink>
      </ThemeProvider>
    );

    const pressableInstance = getByTestId("testOnPress");
    fireEvent.press(pressableInstance);
    expect(onPress).toHaveBeenCalledTimes(1);
  });

  it("doesn't capture the onPress event when the button is disabled ", async () => {
    const onPress = jest.fn();

    const { getByTestId } = await render(
      <ThemeProvider>
        <TextLink onPress={onPress} isDisabled testID="testOnPress">
          Button press
        </TextLink>
      </ThemeProvider>
    );

    const pressableInstance = getByTestId("testOnPress");
    fireEvent.press(pressableInstance);
    expect(onPress).toHaveBeenCalledTimes(0);
  });
});
