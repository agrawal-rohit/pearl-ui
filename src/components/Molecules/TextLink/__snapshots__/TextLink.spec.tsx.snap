// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/TextLink passes the snapshot test for basic setup 1`] = `
<View
  accessibilityHint=""
  accessibilityLabel="Text Link press"
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  onLayout={[Function]}
  style={
    Object {
      "alignSelf": undefined,
      "borderRadius": undefined,
      "bottom": undefined,
      "display": "flex",
      "elevation": undefined,
      "end": undefined,
      "left": undefined,
      "margin": undefined,
      "marginBottom": undefined,
      "marginEnd": undefined,
      "marginHorizontal": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginStart": undefined,
      "marginTop": undefined,
      "marginVertical": undefined,
      "position": undefined,
      "right": undefined,
      "shadowColor": undefined,
      "shadowOffset": undefined,
      "shadowOpacity": undefined,
      "shadowRadius": undefined,
      "start": undefined,
      "top": undefined,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "borderRadius": undefined,
        "display": "flex",
        "overflow": "hidden",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "display": "flex",
            "opacity": 1,
          },
          Object {
            "backgroundColor": undefined,
            "borderBottomColor": undefined,
            "borderBottomWidth": undefined,
            "borderColor": undefined,
            "borderEndColor": undefined,
            "borderEndWidth": undefined,
            "borderLeftColor": undefined,
            "borderLeftWidth": undefined,
            "borderRightColor": undefined,
            "borderRightWidth": undefined,
            "borderStartColor": undefined,
            "borderStartWidth": undefined,
            "borderStyle": undefined,
            "borderTopColor": undefined,
            "borderTopWidth": undefined,
            "borderWidth": undefined,
            "opacity": 1,
          },
        ]
      }
    >
      <Text
        accessibilityRole="text"
        accessible={true}
        allowFontScaling={true}
        scaleFontSize={true}
        style={
          Object {
            "color": "#6A7BFF",
            "display": "flex",
            "fontFamily": "Poppins-Medium",
            "fontSize": 16.099999999999998,
            "fontStyle": "normal",
            "fontWeight": "500",
            "includeFontPadding": false,
            "lineHeight": 22,
          }
        }
        variant="btn2"
      >
        Text Link press
      </Text>
    </View>
  </View>
</View>
`;

exports[`Molecules/TextLink passes the snapshot test for different color schemes 1`] = `
Array [
  <View
    accessibilityHint=""
    accessibilityLabel="Primary text link"
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    onLayout={[Function]}
    style={
      Object {
        "alignSelf": undefined,
        "borderRadius": undefined,
        "bottom": undefined,
        "display": "flex",
        "elevation": undefined,
        "end": undefined,
        "left": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginEnd": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginStart": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
        "start": undefined,
        "top": undefined,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "borderRadius": undefined,
          "display": "flex",
          "overflow": "hidden",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "display": "flex",
              "opacity": 1,
            },
            Object {
              "backgroundColor": undefined,
              "borderBottomColor": undefined,
              "borderBottomWidth": undefined,
              "borderColor": undefined,
              "borderEndColor": undefined,
              "borderEndWidth": undefined,
              "borderLeftColor": undefined,
              "borderLeftWidth": undefined,
              "borderRightColor": undefined,
              "borderRightWidth": undefined,
              "borderStartColor": undefined,
              "borderStartWidth": undefined,
              "borderStyle": undefined,
              "borderTopColor": undefined,
              "borderTopWidth": undefined,
              "borderWidth": undefined,
              "opacity": 1,
            },
          ]
        }
      >
        <Text
          accessibilityRole="text"
          accessible={true}
          allowFontScaling={true}
          scaleFontSize={true}
          style={
            Object {
              "color": "#6A7BFF",
              "display": "flex",
              "fontFamily": "Poppins-Medium",
              "fontSize": 16.099999999999998,
              "fontStyle": "normal",
              "fontWeight": "500",
              "includeFontPadding": false,
              "lineHeight": 22,
            }
          }
          variant="btn2"
        >
          Primary text link
        </Text>
      </View>
    </View>
  </View>,
  <View
    accessibilityHint=""
    accessibilityLabel="Secondary text link"
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    onLayout={[Function]}
    style={
      Object {
        "alignSelf": undefined,
        "borderRadius": undefined,
        "bottom": undefined,
        "display": "flex",
        "elevation": undefined,
        "end": undefined,
        "left": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginEnd": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginStart": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
        "start": undefined,
        "top": undefined,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "borderRadius": undefined,
          "display": "flex",
          "overflow": "hidden",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "display": "flex",
              "opacity": 1,
            },
            Object {
              "backgroundColor": undefined,
              "borderBottomColor": undefined,
              "borderBottomWidth": undefined,
              "borderColor": undefined,
              "borderEndColor": undefined,
              "borderEndWidth": undefined,
              "borderLeftColor": undefined,
              "borderLeftWidth": undefined,
              "borderRightColor": undefined,
              "borderRightWidth": undefined,
              "borderStartColor": undefined,
              "borderStartWidth": undefined,
              "borderStyle": undefined,
              "borderTopColor": undefined,
              "borderTopWidth": undefined,
              "borderWidth": undefined,
              "opacity": 1,
            },
          ]
        }
      >
        <Text
          accessibilityRole="text"
          accessible={true}
          allowFontScaling={true}
          scaleFontSize={true}
          style={
            Object {
              "color": "#FF6AC6",
              "display": "flex",
              "fontFamily": "Poppins-Medium",
              "fontSize": 16.099999999999998,
              "fontStyle": "normal",
              "fontWeight": "500",
              "includeFontPadding": false,
              "lineHeight": 22,
            }
          }
          variant="btn2"
        >
          Secondary text link
        </Text>
      </View>
    </View>
  </View>,
]
`;

exports[`Molecules/TextLink passes the snapshot test for different sizes 1`] = `
Array [
  <View
    accessibilityHint=""
    accessibilityLabel="Text Link press"
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    onLayout={[Function]}
    style={
      Object {
        "alignSelf": undefined,
        "borderRadius": undefined,
        "bottom": undefined,
        "display": "flex",
        "elevation": undefined,
        "end": undefined,
        "left": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginEnd": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginStart": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
        "start": undefined,
        "top": undefined,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "borderRadius": undefined,
          "display": "flex",
          "overflow": "hidden",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        size="xs"
        style={
          Array [
            Object {
              "display": "flex",
              "opacity": 1,
            },
            Object {
              "backgroundColor": undefined,
              "borderBottomColor": undefined,
              "borderBottomWidth": undefined,
              "borderColor": undefined,
              "borderEndColor": undefined,
              "borderEndWidth": undefined,
              "borderLeftColor": undefined,
              "borderLeftWidth": undefined,
              "borderRightColor": undefined,
              "borderRightWidth": undefined,
              "borderStartColor": undefined,
              "borderStartWidth": undefined,
              "borderStyle": undefined,
              "borderTopColor": undefined,
              "borderTopWidth": undefined,
              "borderWidth": undefined,
              "opacity": 1,
            },
          ]
        }
      >
        <Text
          accessibilityRole="text"
          accessible={true}
          allowFontScaling={true}
          scaleFontSize={true}
          style={
            Object {
              "color": "#6A7BFF",
              "display": "flex",
              "fontFamily": "Poppins-Medium",
              "fontSize": 11.5,
              "fontStyle": "normal",
              "fontWeight": "500",
              "includeFontPadding": false,
              "lineHeight": 16,
            }
          }
          variant="btn4"
        >
          Text Link press
        </Text>
      </View>
    </View>
  </View>,
  <View
    accessibilityHint=""
    accessibilityLabel="Text Link press"
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    onLayout={[Function]}
    style={
      Object {
        "alignSelf": undefined,
        "borderRadius": undefined,
        "bottom": undefined,
        "display": "flex",
        "elevation": undefined,
        "end": undefined,
        "left": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginEnd": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginStart": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
        "start": undefined,
        "top": undefined,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "borderRadius": undefined,
          "display": "flex",
          "overflow": "hidden",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        size="s"
        style={
          Array [
            Object {
              "display": "flex",
              "opacity": 1,
            },
            Object {
              "backgroundColor": undefined,
              "borderBottomColor": undefined,
              "borderBottomWidth": undefined,
              "borderColor": undefined,
              "borderEndColor": undefined,
              "borderEndWidth": undefined,
              "borderLeftColor": undefined,
              "borderLeftWidth": undefined,
              "borderRightColor": undefined,
              "borderRightWidth": undefined,
              "borderStartColor": undefined,
              "borderStartWidth": undefined,
              "borderStyle": undefined,
              "borderTopColor": undefined,
              "borderTopWidth": undefined,
              "borderWidth": undefined,
              "opacity": 1,
            },
          ]
        }
      >
        <Text
          accessibilityRole="text"
          accessible={true}
          allowFontScaling={true}
          scaleFontSize={true}
          style={
            Object {
              "color": "#6A7BFF",
              "display": "flex",
              "fontFamily": "Poppins-Medium",
              "fontSize": 13.799999999999999,
              "fontStyle": "normal",
              "fontWeight": "500",
              "includeFontPadding": false,
              "lineHeight": 20,
            }
          }
          variant="btn3"
        >
          Text Link press
        </Text>
      </View>
    </View>
  </View>,
  <View
    accessibilityHint=""
    accessibilityLabel="Text Link press"
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    onLayout={[Function]}
    style={
      Object {
        "alignSelf": undefined,
        "borderRadius": undefined,
        "bottom": undefined,
        "display": "flex",
        "elevation": undefined,
        "end": undefined,
        "left": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginEnd": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginStart": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
        "start": undefined,
        "top": undefined,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "borderRadius": undefined,
          "display": "flex",
          "overflow": "hidden",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        size="m"
        style={
          Array [
            Object {
              "display": "flex",
              "opacity": 1,
            },
            Object {
              "backgroundColor": undefined,
              "borderBottomColor": undefined,
              "borderBottomWidth": undefined,
              "borderColor": undefined,
              "borderEndColor": undefined,
              "borderEndWidth": undefined,
              "borderLeftColor": undefined,
              "borderLeftWidth": undefined,
              "borderRightColor": undefined,
              "borderRightWidth": undefined,
              "borderStartColor": undefined,
              "borderStartWidth": undefined,
              "borderStyle": undefined,
              "borderTopColor": undefined,
              "borderTopWidth": undefined,
              "borderWidth": undefined,
              "opacity": 1,
            },
          ]
        }
      >
        <Text
          accessibilityRole="text"
          accessible={true}
          allowFontScaling={true}
          scaleFontSize={true}
          style={
            Object {
              "color": "#6A7BFF",
              "display": "flex",
              "fontFamily": "Poppins-Medium",
              "fontSize": 16.099999999999998,
              "fontStyle": "normal",
              "fontWeight": "500",
              "includeFontPadding": false,
              "lineHeight": 22,
            }
          }
          variant="btn2"
        >
          Text Link press
        </Text>
      </View>
    </View>
  </View>,
  <View
    accessibilityHint=""
    accessibilityLabel="Text Link press"
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    onLayout={[Function]}
    style={
      Object {
        "alignSelf": undefined,
        "borderRadius": undefined,
        "bottom": undefined,
        "display": "flex",
        "elevation": undefined,
        "end": undefined,
        "left": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginEnd": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginStart": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
        "start": undefined,
        "top": undefined,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "borderRadius": undefined,
          "display": "flex",
          "overflow": "hidden",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        size="l"
        style={
          Array [
            Object {
              "display": "flex",
              "opacity": 1,
            },
            Object {
              "backgroundColor": undefined,
              "borderBottomColor": undefined,
              "borderBottomWidth": undefined,
              "borderColor": undefined,
              "borderEndColor": undefined,
              "borderEndWidth": undefined,
              "borderLeftColor": undefined,
              "borderLeftWidth": undefined,
              "borderRightColor": undefined,
              "borderRightWidth": undefined,
              "borderStartColor": undefined,
              "borderStartWidth": undefined,
              "borderStyle": undefined,
              "borderTopColor": undefined,
              "borderTopWidth": undefined,
              "borderWidth": undefined,
              "opacity": 1,
            },
          ]
        }
      >
        <Text
          accessibilityRole="text"
          accessible={true}
          allowFontScaling={true}
          scaleFontSize={true}
          style={
            Object {
              "color": "#6A7BFF",
              "display": "flex",
              "fontFamily": "Poppins-Medium",
              "fontSize": 18.4,
              "fontStyle": "normal",
              "fontWeight": "500",
              "includeFontPadding": false,
              "lineHeight": 26,
            }
          }
          variant="btn1"
        >
          Text Link press
        </Text>
      </View>
    </View>
  </View>,
]
`;

exports[`Molecules/TextLink passes the snapshot test when using style props 1`] = `
<View
  accessibilityHint=""
  accessibilityLabel="Text link press"
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  onLayout={[Function]}
  style={
    Object {
      "alignSelf": undefined,
      "borderRadius": undefined,
      "bottom": undefined,
      "display": "flex",
      "elevation": undefined,
      "end": undefined,
      "left": undefined,
      "margin": undefined,
      "marginBottom": undefined,
      "marginEnd": undefined,
      "marginHorizontal": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginStart": undefined,
      "marginTop": undefined,
      "marginVertical": undefined,
      "position": undefined,
      "right": undefined,
      "shadowColor": undefined,
      "shadowOffset": undefined,
      "shadowOpacity": undefined,
      "shadowRadius": undefined,
      "start": undefined,
      "top": undefined,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "borderRadius": undefined,
        "display": "flex",
        "overflow": "hidden",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#6A7BFF",
            "display": "flex",
            "opacity": 1,
          },
          Object {
            "backgroundColor": "#6A7BFF",
            "borderBottomColor": undefined,
            "borderBottomWidth": undefined,
            "borderColor": "#C5CEE0",
            "borderEndColor": undefined,
            "borderEndWidth": undefined,
            "borderLeftColor": undefined,
            "borderLeftWidth": undefined,
            "borderRightColor": undefined,
            "borderRightWidth": undefined,
            "borderStartColor": undefined,
            "borderStartWidth": undefined,
            "borderStyle": undefined,
            "borderTopColor": undefined,
            "borderTopWidth": undefined,
            "borderWidth": 2,
            "opacity": 1,
          },
        ]
      }
    >
      <Text
        accessibilityRole="text"
        accessible={true}
        allowFontScaling={true}
        scaleFontSize={true}
        style={
          Object {
            "color": "#6A7BFF",
            "display": "flex",
            "fontFamily": "Poppins-Medium",
            "fontSize": 16.099999999999998,
            "fontStyle": "normal",
            "fontWeight": "500",
            "includeFontPadding": false,
            "lineHeight": 22,
          }
        }
        variant="btn2"
      >
        Text link press
      </Text>
    </View>
  </View>
</View>
`;
